src/main/java/net/snowflake/client/core/QueryContextCache.java

```diff
@@ -152,148 +137,140 @@ public void deserializeFromArrowBase64(String data) {
-        clearCache();
       }
+    }catch (Exception e) {
+      logger.debug("deserializeQueryContextJson: Exception = {}", e.getMessage());
+      // Not rethrowing. clear the cache as incomplete merge can lead to unexpected behavior.
+      clearCache();
+    }
 
-      // After merging all entries, truncate to capacity
-      checkCacheCapacity();
+    // After merging all entries, truncate to capacity
+    checkCacheCapacity();
 
-      logger.debug("deserializeFromArrowBase64() returns");
-      // Log existing cache entries
-      logCacheEntries();
-    } // Synchronized
-  }
+    logger.debug("deserializeQueryContextJson() returns");
+    // Log existing cache entries
+    logCacheEntries();
 
-  /**
-   * Deserialize a query context entry from an Arrow vector
-   *
-   * @param root the Arrow vector root
-   * @param pos the position of the entry to be deserialized from
-   * @return a deserialized query context entry
-   */
-  private static QueryContextElement deserializeEntry(VectorSchemaRoot root, int pos) {
-    BigIntVector idVector = (BigIntVector) root.getVector(ID_POS);
-    BigIntVector tsVector = (BigIntVector) root.getVector(TIMESTAMP_POS);
-    BigIntVector priorityVector = (BigIntVector) root.getVector(PRIORITY_POS);
-    VarBinaryVector contextVector = (VarBinaryVector) root.getVector(CONTEXT_POS);
-
-    return createElement(
-        idVector.get(pos),
-        tsVector.get(pos),
-        priorityVector.get(pos),
-        contextVector.isNull(pos) ? null : contextVector.get(pos));
-  }
+  }// Synchronized
+}
 
-  /**
-   * @param id the id of the element
-   * @param timestamp the last update timestamp
-   * @param priority the priority of the element
-   * @param opaqueContext the binary data of the opaque context
-   * @return a query context element
-   */
-  private static QueryContextElement createElement(
-      long id, long timestamp, long priority, byte[] opaqueContext) {
-    return new QueryContextElement(id, timestamp, priority, opaqueContext);
-  }
+private static QueryContextElement deserializeQueryContextElement(JsonNode node) throws IOException {
+  QueryContextElement entry = new QueryContextElement();
+    JsonNode idNode = node.path("id");
+    if (idNode.isNumber()) {
+        entry.setId(idNode.asLong());
+    }
 
-  /**
-   * Serialize a query context into a base64 encoded string using Arrow. This will be used to send
-   * the serialized query context to Snowflake clients.
-   *
-   * @return If succeeds, the base64 encoded representation of the query context. Otherwise, return
-   *     null.
-   */
-  public String serializeToArrowBase64() {
-    synchronized (this) {
-      logger.debug("serializeToArrowBase64() called");
-      // Log existing cache entries
-      logCacheEntries();
+    JsonNode timestampNode = node.path("timestamp");
+    if (timestampNode.isNumber()) {
+        entry.setReadTimestamp(timestampNode.asLong());
+    }
```

Comment:
what we need to do in "else" part?

Reply:
I think for else part, we simply do nothing and this field will be empty.

Reply:
If entries are not consistent - as in there are type mismatch from what is expected we should skip adding the entire entry to the local cache.

Reply:
Please read Harsh's comment above.

Reply:
I return a `null` and if we obtain a `null` entry, we will skip `merge` it.